### PR TITLE
Improvements of note tests

### PR DESCRIPTION
**Pull request by dglebovskiy**
- Added a descriptive docstring to each test
- Moved test API paths to separate properties
- test_get_detail_json now checks for Unicode field names
- test_put_detail now expects correct note count after the update
- delete-related tests now expect Method Not Allowed response

### CODE REVIEW

File: notes_app/notes/tests.py [hunk 1]
```diff
@@ -27,79 +29,91 @@ def get_credentials(self):
...
         self.assertEqual(Note.objects.count(), 3)
 
         self.assertHttpAccepted(self.api_client.put(
-            '/api/v1/note/1/',
+            self.api_detail,
             format='json',
             data=new_data,
             authentication=self.get_credentials()))
-        self.assertEqual(Note.objects.count(), 4)
+        self.assertEqual(Note.objects.count(), 3)
```
Comment (ghost) [inline notes_app/notes/tests.py:105]: 3 looks like magic number, I think it's better to get count from list endpoint and compare it to `Note.objects.count()`

Comment (ghost) [inline notes_app/notes/tests.py:105]: if it makes sense, of course

**Comment by ghost** [comment]
:+1: for constants !

File: notes_app/notes/tests.py (hunk 1 above)
Comment (dglebovskiy) [reply notes_app/notes/tests.py:105]: Will update in a next commit.

**Pull request merged**